autoload/neomake.vim

```diff
@@ -524,22 +524,31 @@ function! s:command_maker_base._get_tempfilename(jobinfo) abort dict
             let s:pid = getpid()
         endif
         let slash = neomake#utils#Slash()
+
+        let dir = neomake#utils#GetSetting('tempfile_dir', self, '', a:jobinfo.ft, a:jobinfo.bufnr)
+
         let bufname = bufname(a:jobinfo.bufnr)
-        if empty(bufname)
-            let temp_file = tempname() . slash . 'neomaketmp.'.a:jobinfo.ft
-        else
-            let orig_file = neomake#utils#fnamemodify(a:jobinfo.bufnr, ':p')
-            if empty(orig_file)
+
+        let orig_file = neomake#utils#fnamemodify(a:jobinfo.bufnr, ':p')
+        if empty(dir)
+            if empty(bufname)
+                let dir = tempname()
+            elseif empty(orig_file)
                 let dir = tempname()
-                let filename = fnamemodify(bufname, ':t')
-                let s:make_info[make_id].tempfile_dir = dir
             else
                 let dir = fnamemodify(orig_file, ':h')
                 if filewritable(dir) != 2
                     let dir = tempname()
                     let s:make_info[make_id].tempfile_dir = dir
```

Comment:
Related to the above comment: `let s:make_info[make_id].tempfile_dir = dir` got removed above, should this be done later down?